extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java

```diff
@@ -552,4 +555,61 @@ public HistoryHandlerBuildItem(RuntimeValue<Optional<HistoryHandler>> value) {
             this.value = value;
         }
     }
+
+    private static class IDELinksResolver implements ValueResolver {
+
+        private final Path classesPath;
+        private final Path srcMainPath;
+
+        public IDELinksResolver(ArchiveRootBuildItem archiveRoot) {
+            if (archiveRoot.getRootDirs().isSinglePath()) {
+                classesPath = archiveRoot.getRootDirs().getSinglePath();
+                srcMainPath = archiveRoot.getRootDirs().getSinglePath().getParent().getParent()
+                        .resolve("src").resolve("main");
+            } else {
+                classesPath = null;
+                srcMainPath = null;
+            }
+        }
+
+        @Override
+        public boolean appliesTo(EvalContext context) {
+            return ValueResolver.matchClass(context, String.class);
+        }
+
+        @Override
+        public CompletionStage<Object> resolve(EvalContext context) {
+            String className = (String) context.getBase();
+            if ("isAppClass".equals(context.getName())) {
+                if (classesPath == null) {
+                    // we can't really do anything in this case
+                    return CompletableFuture.completedFuture(false);
+                }
+
+                if (className.startsWith("java") || className.startsWith("io.quarkus") || className.startsWith("io.vertx")
+                        || className.startsWith("io.smallrye")) {
+                    // we know these classes are not part of the application so no need to check any deeper
+                    return CompletableFuture.completedFuture(false);
+                }
+
+                // check if the class name corresponds to a file on the file system
+                return CompletableFuture
+                        .completedFuture(Files.exists(classesPath.resolve(className.replace('.', '/') + ".class")));
+            } else if ("ideLink".equals(context.getName())) {
+                if (srcMainPath == null) {
+                    return Results.NOT_FOUND;
+                }
+                // TODO: make it work for non Java source
+                String srcMainJavaPath = srcMainPath.resolve("java").toAbsolutePath().toString().replace('\\', '/');
+
+                // TODO: determine whether IntelliJ or other IDE is being used determine the proper link
+
+                // for this to work, users need to check 'Allow unsigned requests' in the Debugger Settings
+                String link = "http://localhost:63342/api/file/" + srcMainJavaPath + '/'
```

Comment:
This is cool. You can also use `Desktop.getDesktop().open(file)` to invoke the preferred editor in the host machine

Reply:
I didn't know that, thanks!

That is probably going to be a good fallback if we can't determine what IDE is being used (which I hope we can)